checker/tests/nullness/Issue1922.java

```diff
@@ -0,0 +1,32 @@
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import org.checkerframework.checker.nullness.qual.KeyFor;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class Issue1922 {
+    // A method to find a K in the collection and return it, or return null.
+    public static <K> @Nullable K findKey(Collection<@NonNull K> keys, Object target) {
+        for (K key : keys) {
+            if (target.equals(key)) return key;
+        }
+        return null;
+    }
+
+    // Find a key in a map and return String version of its value.
+    public static String findKeyAndFetchString(Map<String, Object> someMap) {
+        // :: error: (type.argument.type.incompatible)
+        @Nullable @KeyFor("someMap") String myKey = Issue1922.<@KeyFor("someMap") String>findKey(someMap.keySet(), "Foo");
+        // If I change above line to this line, I do get an error:
```

Comment:
This comment is misleading, as the above line does give an error.
I would either make the next line a separate test or remove it.